src/stream.rs

```diff
@@ -60,7 +60,7 @@ impl<R: BufRead> Read for Deflate64Decoder<R> {
                 ));
             }
 
-            if result.bytes_written == 0 && !eof && !self.inflater.finished() {
+            if result.bytes_written == 0 && !eof && !self.inflater.finished() && !buf.is_empty() {
```

Comment:
I think it's more readable if we do an early return on empty buf?

Reply:
I agree. switched to early return